gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -131,25 +145,27 @@ public FileTime lastAccessTime() {
 
   @Override
   public Object fileKey() {
-    return metadata.getFilename();
+    return name;
```

Comment:
That's a good question. fileKey should return a canonical name for this file. BlobInfo's documentation just points me to https://cloud.google.com/storage/docs/concepts-techniques#concepts which leaves me unsatisfied. I expect you would know more about this topic than me. Is id the right field to use?